ide/editor.settings.storage/src/org/netbeans/modules/editor/settings/storage/SettingsProvider.java

```diff
@@ -146,39 +146,32 @@ public void propertyChange(PropertyChangeEvent evt) {
             synchronized (this) {
                 boolean fcsChanged = false;
                 boolean kbsChanged = false;
-
-//                if (mimePath.getPath().contains("xml")) {
-//                    System.out.println("@@@ propertyChange: mimePath = " + mimePath.getPath() + " profile = " + fcsProfile + " property = " + evt.getPropertyName() + " oldValue = " + (evt.getOldValue() instanceof MimePath ? ((MimePath) evt.getOldValue()).getPath() : evt.getOldValue()) + " newValue = " + evt.getNewValue());
-//                }
                 
                 // Determine what has changed
+                String prop = evt.getPropertyName();
                 if (this.kbsi == evt.getSource()) {
                     kbsChanged = true;
-                    
-                } else if (evt.getPropertyName() == null) {
+                } else if (prop == null) {
```

Comment:
Just reading the diff, though about using `switch` over `String` in this case?

Reply:
nothing against it, but it would reformat a big section which would make reviews more difficult and might risk to hide the change in it. The main reason I extracted the prop name is because it would make the added `||` longer a few lines later.

Reply:
Yeah, nice to have minimal changes for reviewing.